src/appendbuf.rs

```diff
@@ -248,15 +250,105 @@ impl<'a> AppendBuf<'a> {
   /// ```
   ///
   #[inline]
-  pub fn atomic_slice_pos_from_begin(&self, pos: u32, len: u32) -> Result<AROIobuf, ()> {
+  pub fn atomic_slice_pos_from_begin(&self, pos: u32, len: i64) -> Result<AROIobuf, ()> {
```

Comment:
I wanted to support the python slices model, where if you enter a negative number, it starts counting backwards from the end.  That said, I didn't do it quite right, and should probably just have two functions that support negative numbers in both directions.